src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -484,7 +484,7 @@ void C2_MacroAssembler::fast_lock(Register objReg, Register boxReg, Register tmp
   }
 
 #if INCLUDE_RTM_OPT
-  if (UseRTMForStackLocks && use_rtm) {
+  if (!UseHeavyMonitors && UseRTMForStackLocks && use_rtm) {
```

Comment:
Rather than do this shouldn't `UseHeavyMonitors` and `UseRTMForStackLocks` be mutually exclusive flags, checked in arguments.cpp?